experimental/integrations/prototype-spark/marquez-spark-agent/src/main/java/marquez/spark/agent/lifecycle/SparkSQLExecutionContext.java

```diff
@@ -1,184 +1,206 @@
+      return;
+    }
+
+    LineageEvent event =
+        LineageEvent.builder()
+            .inputs(buildInputs(queryExecution.logical()))
+            .outputs(buildOutputs(queryExecution.logical()))
+            .run(
+                buildRun(
+                    buildRunFacets(
+                        buildLogicalPlanFacet(queryExecution.logical()),
+                        buildPhysicalPlanFacet(queryExecution.executedPlan()),
+                        buildJobErrorFacet(jobEnd.jobResult()))))
+            .job(buildJob())
+            .eventTime(toZonedTime(jobEnd.time()))
+            .eventType(getEventType(jobEnd.jobResult()))
+            .producer("org.apache.spark")
+            .build();
+
+    marquezContext.emit(event);
+  }
+
+  protected String getEventType(JobResult jobResult) {
+    if (jobResult.getClass().getSimpleName().startsWith("JobSucceeded")) {
+      return "COMPLETE";
     }
-    return result;
+    return "FAIL";
   }
 
-  private List<String> findOutputs(LogicalPlan logical) {
-    if (logical instanceof InsertIntoHadoopFsRelationCommand) {
-      InsertIntoHadoopFsRelationCommand insert = (InsertIntoHadoopFsRelationCommand)logical;
-      return ImmutableList.of(insert.outputPath().toUri().toString());
-    } else {
-      extraInfo.append("No Output found for plan: ").append(logical.getClass().getName()).append(" ").append(logical).append("\n");
-      return ImmutableList.of();
+  private List<Dataset> buildInputs(LogicalPlan logical) {
+    List<Dataset> inputDatasets = new ArrayList<>();
+    Collection<LogicalPlan> leaves = asJavaCollection(logical.collectLeaves());
+    for (LogicalPlan leaf : leaves) {
+      if (!(leaf instanceof LogicalRelation)) {
+        continue;
+      }
+      LogicalRelation lr = (LogicalRelation) leaf;
+      BaseRelation lrr = lr.relation();
+      if (!(lrr instanceof HadoopFsRelation)) {
+        continue;
+      }
+      FileIndex location = ((HadoopFsRelation) lrr).location();
+      Collection<Path> rootPaths = asJavaCollection(location.rootPaths());
+      for (Path rootPath : rootPaths) {
+        Dataset.DatasetBuilder lineageDataset =
+            Dataset.builder()
+                .namespace(marquezContext.getJobNamespace())
+                .name(rootPath.toUri().toString().replaceAll(":", "_"));
+        inputDatasets.add(lineageDataset.build());
+      }
     }
+
+    return inputDatasets;
   }
 
-  private void print(String prefix, SparkPlanInfo i) {
-    extraInfo.append(prefix + i.nodeName()+ "(" + i.simpleString()+ " metadata: " + i.metadata() + " metrics: "+ i.metrics() + ")" ).append("\n");
-    Collection<SparkPlanInfo> children = JavaConversions.asJavaCollection(i.children());
-    for (SparkPlanInfo child : children) {
-      print(prefix + "  ", child);
+  private List<Dataset> buildOutputs(LogicalPlan logical) {
+    List<Dataset> inputDatasets = new ArrayList<>();
+    if (logical instanceof InsertIntoHadoopFsRelationCommand) {
```

Comment:
same comment for unknown outputs, we should have an else case that tracks the type and content of unknown commands.

Reply:
The logical plan facet attempts to log unknown datasets